### DISCUSSION TITLE
vi-mode: breaks back deletion

### DISCUSSION THREAD

**Discussion by Integralist** [ghostty-org/ghostty#3487] [Feature Requests, Ideas] [answered]
👋

I have vi-mode enabled in my zsh shell (macOS):

```shell
setopt VI
```

If I type a command like...

```shell
ls -la
```

...then press `ESC` to go into VI mode, and then press `I` to go back to normal edit mode, then I find that I can no longer delete characters using the back-delete button. The cursor line stays at the end of the input.

See the video, which hopefully demonstrates the issue:
https://github.com/user-attachments/assets/5fafdf3b-7ca5-4704-9955-eb43349a91c9

**Comment by mitchellh** [comment]
Convert to discussion for issue triage.

**Comment by ferdinandrau** [comment, marked as answer]
This is the expected behavior of Zsh, try the same in other terminals, the same thing happens. 

Solution:

> The quick answer, put this in your ~/.zshrc:
  ```
  bindkey -v '^?' backward-delete-char
  ```
> Normally '^?' (backspace) is bound to vi-backward-delete-char, which is supposed to have the behavior you observe, as that is how vi (not vim, unless in vi classic mode) behaves.


Taken from: https://unix.stackexchange.com/questions/290392/backspace-in-zsh-stuck?rq=1
